micrometer-core/src/main/java/io/micrometer/core/instrument/binder/db/PostgreSQLDatabaseMetrics.java

```diff
@@ -366,4 +408,59 @@ private Names() {
 
     }
 
+    @FunctionalInterface
+    interface ResultSetGetter<T> {
+
+        T get(ResultSet resultSet) throws SQLException;
+
+    }
+
+    static final class Version {
+
+        private static final Pattern VERSION_PATTERN = Pattern.compile("^(\\d+\\.\\d+).*");
+
+        static final Version EMPTY = new Version(0, 0);
+
+        static final Version V17 = new Version(17, 0);
+
+        final int majorVersion;
+
+        final int minorVersion;
+
+        static Version parse(String versionString) {
+            try {
+                final Matcher matcher = VERSION_PATTERN.matcher(versionString);
+                if (!matcher.matches()) {
+                    return EMPTY;
```

Comment:
Are there expected cases in which the version string the server returns won't be parseable by this method? I'm wondering if we should log something when parsing the version fails, or perhaps when `getServerVersion` returns `Version.EMPTY`. Otherwise, users may not notice the issue early on and it will delay feedback about a situation we potentially need to fix this code to handle. What do you think?